src/libraries/System.Private.DataContractSerialization/src/System/Runtime/Serialization/ClassDataContract.cs

```diff
@@ -711,7 +729,13 @@ private sealed class ClassDataContractCriticalHelper : DataContract.DataContract
             public XmlDictionaryString[]? MemberNames;
             public XmlDictionaryString[]? MemberNamespaces;
 
-            internal ClassDataContractCriticalHelper(Type type) : base(type)
+            [RequiresUnreferencedCode(DataContractJsonSerializer.SerializerTrimmerWarning)]
+            internal ClassDataContractCriticalHelper([DynamicallyAccessedMembers(
+                    DynamicallyAccessedMemberTypes.PublicMethods |
```

Comment:
1. No Properties or Fields?
2. Can we store this in a const somewhere so we don't need to repeat it?

Reply:
1. Good catch, I just wasn't getting the warning for those since I had already added the RUC attribute. Added.
2. Fixed.